src/pages/ProcessJoinTeam.vue

```diff
@@ -0,0 +1,183 @@
+ */
+import { gql } from '@apollo/client';
+
+import WwwPage from '@/components/WwwFrame/WwwPage';
+import joinTeam from '@/graphql/mutation/joinTeam.graphql';
+import KvPageContainer from '~/@kiva/kv-components/vue/KvPageContainer';
+import KvGrid from '~/@kiva/kv-components/vue/KvGrid';
+import KvLoadingSpinner from '~/@kiva/kv-components/vue/KvLoadingSpinner';
+
+const userTeamMembership = gql`query userTeamMembership( $teamPublicId: String!) {
+	my {
+		id
+		userAccount {
+		id
+		}
+	}
+	community {
+		team(teamPublicId: $teamPublicId) {
+			id
+			userProperties {
+				membershipStatus
+			}
+			name
+		}
+	}
+}`;
+
+export default {
+	name: 'ProcessJoinTeam',
+	props: {
+		doneUrl: {
+			type: String,
+			default: null
+		},
+		teamRecruitmentId: {
+			type: Number,
+			default: null,
+		},
+		teamPublicId: {
+			type: String,
+			required: true
+		},
+		promoId: {
+			type: Number,
+			default: null
+		},
+	},
+	components: {
+		KvLoadingSpinner,
+		KvGrid,
+		KvPageContainer,
+		WwwPage,
+	},
+	inject: ['apollo', 'cookieStore'],
+	apollo: {
+		preFetch: true,
+		query: userTeamMembership,
+		preFetchVariables({ route }) {
+			return {
+				teamPublicId: route.query.teamPublicId,
+			};
+		},
+		variables() {
+			return {
+				teamPublicId: this.teamPublicId,
+			};
+		},
+		result({ data }) {
+			this.userMembershipStatus = data?.community?.team?.userProperties?.membershipStatus ?? 'none';
+			this.teamName = data?.community?.team?.name ?? '';
+			this.teamId = data?.community?.team?.id ?? 0;
+		}
+	},
+	data() {
+		return {
+			userMembershipStatus: 'none',
+			isLoading: true,
+			teamName: '',
+		};
+	},
+	computed: {
+		isPending() {
+			return this.userMembershipStatus === 'requestPending';
+		},
+		isMember() {
+			return this.userMembershipStatus === 'member' || this.userMembershipStatus === 'captain';
+		},
+		isNotMember() {
+			return this.userMembershipStatus === 'none'
+			|| this.userMembershipStatus === 'recruited'
+			|| this.userMembershipStatus === 'recruitedByPromo';
+		},
+	},
+	methods: {
+		memberRedirect() {
+			if (!this.$isServer) {
+				window.location = this.doneUrl ? this.doneUrl : `/team/${this.teamPublicId}`;
+			}
+		},
+		async handleJoinTeam() {
+			try {
+				await this.apollo.mutate({
+					mutation: joinTeam,
+					variables: {
+						team_id: this.teamId,
+						team_recruitment_id: this.teamRecruitmentId,
+						promo_id: this.promoId
+					}
+				});
+				// get updated team membership status to determine if joined or pending
+				const { data: userMembershipData } = await this.apollo.query({
+					query: userTeamMembership,
+					variables: {
+						teamPublicId: this.teamPublicId,
+					},
+					fetchPolicy: 'network-only'
+				});
+				// update the users membership status
+				this.userMembershipStatus = userMembershipData?.community?.team?.userProperties?.membershipStatus;
+				if (this.isMember) {
+					this.$showTipMsg('You have joined the team!');
+					this.memberRedirect();
```

Comment:
Do we want a slight delay before redirect so that the user sees the toast?

Reply:
I was thinking the same! Delay 5-10 secs would at least allow someone to scan/read the page.

Reply:
I considered that approach, I think its fine, the delay will be the page loading :)